console/src/components/Services/Data/TableModify/ComputedFieldsEditor.js

```diff
@@ -0,0 +1,409 @@
+
+        const newState = [...stateComputedFields];
+
+        newState[i] = {
+          ...newState[i],
+          definition: {
+            ...newState[i].definition,
+            function: {
+              ...newState[i].definition.function,
+              schema: e.target.value,
+            },
+          },
+        };
+
+        setComputedFieldsState(newState);
+      };
+
+      const handleFnNameChange = e => {
+        const newState = [...stateComputedFields];
+
+        newState[i] = {
+          ...newState[i],
+          definition: {
+            ...newState[i].definition,
+            function: {
+              ...newState[i].definition.function,
+              name: e.target.value,
+            },
+          },
+        };
+
+        setComputedFieldsState(newState);
+      };
+
+      const handleCommentChange = e => {
+        const newState = [...stateComputedFields];
+
+        newState[i] = {
+          ...newState[i],
+          comment: e.target.value,
+        };
+
+        setComputedFieldsState(newState);
+      };
+
+      const handleTableRowArgChange = e => {
+        const newState = [...stateComputedFields];
+
+        newState[i] = {
+          ...newState[i],
+          definition: {
+            ...newState[i].definition,
+            table_argument: e.target.value || null,
+          },
+        };
+
+        setComputedFieldsState(newState);
+      };
+
+      return (
+        <div>
+          <div>
+            <div className={`${styles.add_mar_bottom_mid}`}>
+              <b>Computed field name:</b>
+            </div>
+            <input
+              type="text"
+              value={computedFieldName}
+              onChange={handleNameChange}
+              className={`form-control ${styles.wd50percent}`}
+            />
+          </div>
+          <div className={`${styles.add_mar_top}`}>
+            <div className={`${styles.add_mar_bottom_mid}`}>
+              <b>Function schema: </b>
+            </div>
+            <select
+              value={computedFieldFunctionSchema}
+              className={`${styles.select} form-control ${
+                styles.add_pad_left
+              } ${styles.wd50percent}`}
+              data-test={'computed_field-fn-ref-schema'}
+              onChange={handleFnSchemaChange}
+            >
+              {schemaList.map((s, j) => {
+                const schemaName = getSchemaName(s);
+
+                return (
+                  <option key={j} value={schemaName}>
+                    {schemaName}
+                  </option>
+                );
+              })}
+            </select>
+          </div>
+          <div className={`${styles.add_mar_top}`}>
+            <div className={`${styles.add_mar_bottom_mid}`}>
+              <b>Function name: </b>
+              <RawSqlButton
+                dataTestId={'create-function'}
+                customStyles={`${styles.display_inline} ${styles.add_mar_left}`}
+                sql={''}
```

Comment:
Can we pass a template function here?
For people who are trying out and don't know the exact syntax of SQL functions (peeps like me). Also for good demos.

Reply:
I has though about this but there is no generic template I could come up with that can be used without making assumptions about the function. The create function structure is defined [here](https://www.postgresql.org/docs/current/sql-createfunction.html) and you can see it cant really be templated without making  assumptions about return types and language